### PR TITLE
fix: set_cookie() does not use Config\Cookie values

### LINKED ISSUE #6540: Bug: set_cookie not using default value of $secure from config
### PHP Version

7.4

### CodeIgniter4 Version

4.2.6

### CodeIgniter4 Installation Method

Manual (zip or tar.gz)

### Which operating systems have you tested for this bug?

macOS

### Which server did you use?

apache

### Database

_No response_

### What happened?

set_cookie function in Helpers/cookie_helper.php is not using the default value $secure i specified in app/Config/Cookie.php. 

This used to work in CI3 because I think the default value of $secure param of set_cookie function used to be null which later on fallback to default.****

### Steps to Reproduce

1. Set public $secure = true in /app/Config/Cookie.php'
2. Call the helper function set_cookie
   ```php
        helper('cookie');
        set_cookie([
            'name' => 'pmw',
            'value' => 2,
            'samesite' => 'None',
        ]);
   ```
4. CookieException::forInvalidSameSiteNone() is thrown although $secure was suppose to be true

### Expected Output

No exception

### Anything else?

_No response_

### PR DESCRIPTION
**Pull request by kenjis**
**Description**
Fixes #6540

**Checklist:**
- [x] Securely signed commits
- [x] Component(s) with PHPDoc blocks, only if necessary or adds value
- [ ] Unit testing, with >80% coverage
- [x] User guide updated
- [x] Conforms to style guide

### CODE REVIEW

**Review by MGatner — APPROVED** [review]

File: system/HTTP/ResponseTrait.php [hunk 1]
```diff
@@ -567,8 +568,17 @@ public function setCookie(
             return $this;
         }
 
+        /** @var CookieConfig|null $cookieConfig */
+        $cookieConfig = config('Cookie');
+
+        if ($cookieConfig) {
```
Comment (samsonasik) [inline system/HTTP/ResponseTrait.php:574]: using instanceof is better  I think, that's we know that type is exactly a CookieConfig:

```suggestion
        if ($cookieConfig instanceof CookieConfig) {
```

File: tests/system/HTTP/ResponseCookieTest.php [hunk 2]
```diff
@@ -255,6 +256,7 @@ public function testCookieStrictSetSameSite()
 
     public function testCookieBlankSetSameSite()
     {
+        /** @var \Config\Cookie $config */
```
Comment (samsonasik) [inline tests/system/HTTP/ResponseCookieTest.php:259]: there is alias for it, so use `CookieConfig` for shorter:

```suggestion
        /** @var CookieConfig $config */
```

File: tests/system/HTTP/ResponseCookieTest.php [hunk 3]
```diff
@@ -314,6 +316,30 @@ public function testCookieInvalidSameSite()
         ]);
     }
 
+    public function testSetCookieConfigCookieIsUsed()
+    {
+        /** @var \Config\Cookie $config */
```
Comment (samsonasik) [inline tests/system/HTTP/ResponseCookieTest.php:321]: there is alias for it, so use `CookieConfig` for shorter:

```suggestion
        /** @var CookieConfig $config */
```

File: tests/system/HTTP/ResponseCookieTest.php (hunk 2 above)
Comment (kenjis) [reply tests/system/HTTP/ResponseCookieTest.php]: Done.

File: tests/system/HTTP/ResponseCookieTest.php (hunk 3 above)
Comment (kenjis) [reply tests/system/HTTP/ResponseCookieTest.php]: Done.

**Review by samsonasik — APPROVED** [review]

**Pull request merged**